src/pooler.erl

```diff
@@ -392,8 +392,10 @@ handle_info(timeout, #pool{group = Group} = Pool) ->
 handle_info({'DOWN', MRef, process, Pid, Reason}, State) ->
     State1 =
         case dict:find(Pid, State#pool.all_members) of
+            %% the connection is not alive
             {ok, {_PoolName, _ConsumerPid, _Time}} ->
                 do_return_member(Pid, fail, State);
+            %% connection is not alive, look for the consumer
```

Comment:
This comment is a bit off.  This case is when the process that has gone down is not a known member of the pool and thus we see if the process is a _consumer_ of the pool.